### ISSUE TITLE
Should Literal account Final strings?

### ISSUE THREAD

**Issue by dimaqq** [microsoft/pyright#7185]
Consider this code:
```py
from typing import Final, Literal


CAT: Final = "cat"
DOG: Final = "dog"
PETS = Literal[CAT, DOG]
```

`pyright 1.1.349` considers this invalid:
```
> pyright pets.py
/.../pets.py
  /.../pets.py:6:16 - error: Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value
  /.../pets.py:6:21 - error: Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value
2 errors, 0 warnings, 0 informations
```

I wonder if this should be valid... on one hand, `CAT` is not a literal string, but then it's value is hinted not change 🤔 

Is there a canonical way to write what I mean?

`Final` was suggested here: https://github.com/python/mypy/issues/16791

**Comment by erictraut** [comment]
Pyright is behaving correctly here, so I don't consider this a bug. According to the Python typing spec, you need to use literal expressions within a `Literal` type annotation.
